### PR TITLE
Drop redundant legacy API endpoints 

### PR DESCRIPTION
**Pull request by mvdbeek**
Builds on #13579 to identify what needs to be dropped from buildapp.py.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by davelopez** [comment]
It seems there are some legacy endpoints that bypass the FastAPI ones. The `lib/galaxy_test/api/test_api_batch.py::ApiBatchTestCase::test_querystring_params` bite me too, I think `BatchMiddleware` is still somewhat using the old routes

**Comment by mvdbeek** [comment]
Oh, nice catch, yes, we can probably drop the middleware

File: lib/galaxy/webapps/galaxy/api/__init__.py [hunk 1]
```diff
@@ -257,6 +257,10 @@ def get(self, *args, **kwd):
         """Extend FastAPI.get to accept a require_admin Galaxy flag."""
         return super().get(*args, **self._handle_galaxy_kwd(kwd))
 
+    def patch(self, *args, **kwd):
+        """Extend FastAPI.put to accept a require_admin Galaxy flag."""
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/api/__init__.py]:
```suggestion
        """Extend FastAPI.patch to accept a require_admin Galaxy flag."""
```

File: lib/galaxy/webapps/galaxy/buildapp.py [hunk 2]
```diff
@@ -460,15 +334,14 @@ def populate_api_routes(webapp, app):
     )
     webapp.mapper.connect("/api/upload/resumable_upload", controller="uploads", action="hooks")
     webapp.mapper.connect("/api/upload/hooks", controller="uploads", action="hooks", conditions=dict(method=["POST"]))
-    webapp.mapper.connect("/api/ftp_files", controller="remote_files")
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/buildapp.py:463]: Seems that `/api/ftp_files` is missing. Was it an alias for `/api/remote_files`?

**Review by davelopez — APPROVED** [review]
Everything looks good! Only found a missing route that may be an alias.

Thank you so much also for the fixes and cleaning of unused routes!

File: lib/galaxy/webapps/galaxy/buildapp.py (hunk 2 above)
Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/buildapp.py:463]: Yep. I'll restore the alias

**Pull request merged**